src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportManager.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Net.Quic;
+using System.Runtime.Versioning;
+using System.Net.Http.Headers;
+using System.Net.Security;
+using System.Text;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    [SupportedOSPlatform("windows")]
+    [SupportedOSPlatform("linux")]
+    [SupportedOSPlatform("macos")]
+    internal sealed class Http3WebtransportManager : IDisposable, IAsyncDisposable
+    {
+        private ConcurrentDictionary<long, Http3WebtransportSession> _sessions;
+        private QuicConnection _connection;
+
+        public Http3WebtransportManager(QuicConnection connection)
+        {
+            _sessions = new ConcurrentDictionary<long, Http3WebtransportSession>();
+            _connection = connection;
+        }
+
+        public bool AddSession(QuicStream connectStream, Http3WebtransportSession webtransportSession)
+        {
+            bool ans = _sessions.TryAdd(connectStream.Id, webtransportSession);
```

Comment:
What do you think it would mean if TryAdd returns `false`?

Reply:
Placed inside lock and added debug assert for boolean